File: data_reciever.js
const net = require('net');

// Define the host and port of the socket
const HOST = 'localhost';
const PORT = 8000;

// Create a socket object
const server = net.createServer((socket) => {
  console.log(`Connected to ${socket.remoteAddress}:${socket.remotePort}`);

  // Listen for incoming data
  socket.on('data', (data) => {
    // Convert the data to a JavaScript object
    const data_obj = JSON.parse(data.toString('utf-8'));

    // Process the data however you need to
    const camera_id = data_obj.camera_id;
    const student_strength = data_obj.student_strength;
    console.log(`Camera ID: ${camera_id}, Student Strength: ${student_strength}`);
  });

  // Handle socket errors
  socket.on('error', (err) => {
    console.error(`Socket error: ${err.message}`);
  });

  // Handle socket closure
  socket.on('close', () => {
    console.log('Socket closed');
  });
});

// Start the server and listen for incoming connections
server.listen(PORT, HOST, () => {
  console.log(`Server listening on ${HOST}:${PORT}`);
});

File: backend/data_reciever.py
import redis
import json

r = redis.Redis(host='localhost', port=6379, db=0)
p = r.pubsub()
p.subscribe('student_count')

strength = {}
count = {}


def update_average(key, value):
    if key not in strength:
        strength[key] = value
        count[key] = 1
    else:
        strength[key] += value
        count[key] += 1


def get_average(key):
    return int(strength[key] / count[key])


def reset():
    for i in count:
        count[i] = 1


for message in p.listen():

    if message['type'] == 'message':
        data = json.loads(message['data'].decode())
        for camera in data:
            update_average(camera, data[camera])

    total_strength = 0
    for key in strength:
        avg = get_average(key)
        total_strength+=avg
        # print(key, " ", avg)
    print(total_strength)